pkg/utils/spiffe/spiffe.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spiffe
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	b64 "encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+
+	nfdv1alpha1 "sigs.k8s.io/node-feature-discovery/api/nfd/v1alpha1"
+)
+
+type SpiffeObject struct {
+	Spec      nfdv1alpha1.NodeFeatureSpec
+	Name      string
+	Namespace string
+	Labels    map[string]string
+}
+
+// WorkerSpiffeID is the SpiffeID of the worker
+const WorkerSpiffeID = "spiffe://nfd.k8s-sigs.io/worker"
+
+type SpiffeClient struct {
+	WorkloadApiClient workloadapi.Client
+}
+
+var latestSignature = []byte{}
+
+func NewSpiffeClient(socketPath string) (*SpiffeClient, error) {
+	spiffeClient := SpiffeClient{}
+	workloadApiClient, err := workloadapi.New(context.Background(), workloadapi.WithAddr(socketPath))
+	if err != nil {
+		return nil, err
+	}
+	spiffeClient.WorkloadApiClient = *workloadApiClient
+	return &spiffeClient, nil
+}
+
+func SignData(data SpiffeObject, privateKey crypto.Signer) ([]byte, error) {
+	stringifyData, err := json.Marshal(data)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	dataHash := sha256.Sum256([]byte(stringifyData))
+	if len(latestSignature) > 0 {
```

Comment:
We'd still need to check the hash too, right? Only return the cached signature if the cached hash matches.